workers/securityworker/securityworker.py

```diff
@@ -29,7 +30,15 @@ def __init__(self):
         self.add_operation(self._index_in_scanner, interval)
 
     def _index_in_scanner(self):
-        self._next_token = self._model.perform_indexing(self._next_token)
+        try:
+            with GlobalLock(self._next_token.normalize(), lock_ttl=DEFAULT_INDEXING_INTERVAL + 60):
```

Comment:
How does this TTL work? If it's only needed to adjudicate race and not while the operation is in flight, this is probably fine. If it's needed for the whole operation, this is probably too low. 90sec is probably one request out of the batch.

Reply:
Ideally it should be as long as the indexing operation lasts for that batch. Otherwise, you run into the possibility that another worker might try and do duplicative work on the same batch.

Reply:
So, with a batch size of ~30k (seen on quayio) and response time of 10 sec (eyeballing the latency information from quayio's clair) means the timeout should be about 83 hours.

Reply:
`DEFAULT_INDEXING_INTERVAL` is set to 30s by default.  From the above what I am understanding is that we will try to acquire a lock and if we get blocked we will wait for the TTL. If we cannot acquire that lock, we will assume that this block is being worked on by some other woker and proceed to acquire then next batch of work.

Reply:
@syed Yes. If a worker cannot acquire a lock, it should try on the next "page". 
@hdonnay I updated the lock library, to one that supports auto renewal, as long as the context hasn't finished.